pkg/lib/tmp/copy.go

```diff
@@ -0,0 +1,49 @@
+package tmp
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
```

Comment:
I don't think it's worth aggregating errors, as long as we return the first error that was encountered.

Reply:
I agree, it looks like any error in this codepath is terminal for the registry